src/plugins/csvstorage/csvstorage.c

```diff
@@ -31,7 +31,13 @@ static char *parseLine(char *origLine, char delim, unsigned long offset)
 
 	if(ptr == NULL)
 	{
-		line[elektraStrLen(line)-2]='\0';
+		ssize_t len = elektraStrLen(line);
+		if(line[len-2] == '\n')
+			line[len-2] = '\0';
+		else if(line[len-1] == '\0')
+		{
+			ELEKTRA_ADD_WARNINGF(136, parentKey, "Unexpected end of line, expected \\n, got \\0 \n %s\n", origLine);
+		}
```

Comment:
else missing, what if its not \n, but \r\f,..? There should be at least an else stating that an unknown end of line was received.

Please add line numbers more visible. (in line ...)